### PR TITLE
Adding prefix to share source. [ENG-2115]

### PR DESCRIPTION
**Pull request by corbinSanders**
## Purpose

Adding a prefix to the share source name

## Changes

Modifying setup_share_source. Adding tests

## QA Notes

  - Does this change require a data migration? If so, what data will we migrate? N/A
  - What is the level of risk? Minimal, just for staging environments
    - Any permissions code touched? no
    - Is this an additive or subtractive change, other? additive
  - How can QA verify? (Through UI, API, AdminApp or AdminAdminApp?) UI/Admin app
    - If verifying through API, what's the new version? Please include the endpoints in PR notes or Dev docs. N/A
  - What features or workflows might this change impact? Provider creation. And share
  - How will this impact performance? Shouldn't
-->

## Documentation

N/A

## Side Effects

Shouldn't be.
## Ticket

https://openscience.atlassian.net/browse/ENG-2115

### CODE REVIEW

File: admin_tests/preprint_providers/test_views.py [hunk 1]
```diff
@@ -88,6 +90,35 @@ def test_update_share_token_and_source(self):
         assert self.preprint_provider.access_token == token
         assert self.preprint_provider.share_source == source_name
 
+    @responses.activate
+    @mock.patch('api.share.utils.settings.SHARE_ENABLED', True)
+    @mock.patch.object(settings, 'SPAM_FLAGGED_MAKE_NODE_PRIVATE', 'testenv')
```
Comment (brianjgeiger) [inline admin_tests/preprint_providers/test_views.py:95]: Drive-by: Should this be `SHARE_PROVIDER_PREPEND`?

Comment (brianjgeiger) [reply admin_tests/preprint_providers/test_views.py:95]: Also, if so, how did tests pass?

Comment (Johnetordoff) [reply admin_tests/preprint_providers/test_views.py:95]: I believe this actually *should* pass, but only because the response mocking includes `testenv_`, mocking `SPAM_FLAGGED_MAKE_NODE_PRIVATE` as `testenv` is just a repetitive naming choice. I think if you refactor to `@mock.patch.object(settings, 'SPAM_FLAGGED_MAKE_NODE_PRIVATE', True)` it will still work. 

The thing this test is missing though is that you don't assert that `longTitle` is sent properly in the request, only that it's received in the response properly causing some of this assertion confusion.

File: admin_tests/registration_providers/test_views.py [hunk 2]
```diff
@@ -166,6 +167,34 @@ def test_share_source(self, mock_share, view, provider, req):
         assert provider.share_source == 'test source'
         assert provider.access_token == 'test access token'
 
+    @mock.patch.object(settings, 'SPAM_FLAGGED_MAKE_NODE_PRIVATE', 'testenv')
+    def test_share_source_prefix(self, mock_share, view, provider, req):
```
Comment (Johnetordoff) [inline admin_tests/registration_providers/test_views.py:171]: You should check the requests in this test too.  `responses.calls[-1].request` for the most recent request.
